modules/cbMap/processmap/importCSV.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+//ini_set('display_errors', 'on');
+require_once("config.inc.php");
+include_once('data/CRMEntity.php');
+include_once('modules/cbMap/cbMap.php');
+require_once('include/utils/utils.php');
+require_once('include/database/PearDatabase.php');
+
+global $adb, $log, $root_directory, $current_user;
+$current_user = new Users();
+$current_user->retrieveCurrentUserInfoFromFile(1);
+if (isset($argv) && !empty($argv)) {
+    $csvfile = $argv[1];
+    $mapid = $argv[2];
+}
+$csvfile=explode("=",$csvfile);
+$mp=explode("=",$mapid);
+$mapid=$mp[1];
+$filename = "import/$csvfile[1]";
+$table = pathinfo($filename);
+
+$tb=explode("=",$table['filename']);
+$table = "massivelauncher_" . $tb[0];
+$drop = "drop table if exists $table;";
+$adb->query($drop);
+$delimiter = ',';
+
+$fp = fopen($filename, 'r');
+$frow = fgetcsv($fp, 1000, $delimiter);
+
+$allHeaders = implode(",", $frow);
+$columns = "`id` INT(11) NOT NULL AUTO_INCREMENT PRIMARY KEY,
+          `selected` varchar(3) ";
+foreach ($frow as $column) {
+if($column=='') $column='lastvalue';
+    $columns .= ", `$column` varchar(250)";
+}
+$create = "create table if not exists $table ($columns);";
+$adb->query($create);
+
+$file = $root_directory . $filename;
+$irow=0;
+while (($data = fgetcsv($fp, 1000, $delimiter)) !== FALSE) {
+        $row_vals=implode("','",$data);
+        $str="INSERT INTO $table  VALUES ('','','$row_vals')";
+        echo $str;
+        $adb->query($str);//break;
+    $irow++;
+    }
+$mapfocus = CRMEntity::getInstance("cbMap");
+$mapfocus->retrieve_entity_info($mapid, "cbMap");
+$mapinfo = $mapfocus->readImportType();
+$updateFld = $mapinfo['target'];
+$matchFld = $mapinfo['match'];
+$options = $mapinfo['options'];
+
+$module = $mapfocus->getMapTargetModule();
+include_once("modules/$module/$module.php");
```

Comment:
this line is not needed, CRMEntity::getInstance below checks the file exists and includes it for you